### PR TITLE
only show columns for selected type

### PR DESCRIPTION
**Pull request by snopoke**
http://manage.dimagi.com/default.asp?224209
@czue

### CODE REVIEW

File: corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js [hunk 1]
```diff
@@ -1037,7 +1043,8 @@ hqDefine('app_manager/js/detail-screen-config.js', function () {
                         displayColumn: spec.fixtureSelect.display_column,
                         localize: spec.fixtureSelect.localize,
                         variableColumn: spec.fixtureSelect.variable_column,
-                        xpath: spec.fixtureSelect.xpath
+                        xpath: spec.fixtureSelect.xpath,
+                        fixture_columns_by_type: spec.fixture_columns_by_type
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js]: Missing trailing comma. (comma-dangle)

File: corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js [hunk 2]
```diff
@@ -416,6 +416,12 @@ hqDefine('app_manager/js/detail-screen-config.js', function () {
         self.localize = ko.observable(init.localize);
         self.variableColumn = ko.observable(init.variableColumn);
         self.xpath = ko.observable(init.xpath);
+        self.fixture_columns = ko.computed(function() {
+            var columns_for_type = init.fixture_columns_by_type[self.fixtureType()],
+                default_option = [gettext("Select One")];
+            return default_option.concat(columns_for_type)
+        })
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js]: Missing semicolon. (semi)

File: corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js [hunk 3]
```diff
@@ -416,6 +416,12 @@ hqDefine('app_manager/js/detail-screen-config.js', function () {
         self.localize = ko.observable(init.localize);
         self.variableColumn = ko.observable(init.variableColumn);
         self.xpath = ko.observable(init.xpath);
+        self.fixture_columns = ko.computed(function() {
+            var columns_for_type = init.fixture_columns_by_type[self.fixtureType()],
+                default_option = [gettext("Select One")];
+            return default_option.concat(columns_for_type)
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js]: Missing semicolon. (semi)

File: corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js [hunk 4]
```diff
@@ -416,6 +416,12 @@ hqDefine('app_manager/js/detail-screen-config.js', function () {
         self.localize = ko.observable(init.localize);
         self.variableColumn = ko.observable(init.variableColumn);
         self.xpath = ko.observable(init.xpath);
+        self.fixture_columns = ko.computed(function() {
+            var columns_for_type = init.fixture_columns_by_type[self.fixtureType()],
+                default_option = [gettext("Select One")];
+            return default_option.concat(columns_for_type);
+        })
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js]: Missing semicolon. (semi)

**Comment by czue** [comment]
I don't have much context for this change but looks good to me. is there anyone else better to review?

**Pull request merged**